cmd/aws-actuator/utils.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"os/exec"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/golang/glog"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/framework"
+	machineactuator "sigs.k8s.io/cluster-api-provider-aws/pkg/actuators/machine"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1alpha1"
+	awsclient "sigs.k8s.io/cluster-api-provider-aws/pkg/client"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+type manifestParams struct {
+	ClusterID string
+}
+
+func readMachineManifest(manifestParams *manifestParams, manifestLoc string) (*clusterv1.Machine, error) {
+	machine := &clusterv1.Machine{}
+	manifestBytes, err := ioutil.ReadFile(manifestLoc)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read %v: %v", manifestLoc, err)
+	}
+
+	t, err := template.New("machineuserdata").Parse(string(manifestBytes))
+	if err != nil {
+		return nil, err
+	}
+	var buf bytes.Buffer
+	err = t.Execute(&buf, *manifestParams)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = yaml.Unmarshal(buf.Bytes(), &machine); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal %v: %v", manifestLoc, err)
+	}
+
+	return machine, nil
+}
+
+func readClusterResources(manifestParams *manifestParams, clusterLoc, machineLoc, awsCredentialSecretLoc, userDataLoc string) (*clusterv1.Cluster, *clusterv1.Machine, *apiv1.Secret, *apiv1.Secret, error) {
+	var err error
+	machine, err := readMachineManifest(manifestParams, machineLoc)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	cluster := &clusterv1.Cluster{}
+	{
```

Comment:
Historical reasons. In the past there were more object initialization. The purpose was to reuse the same variables since all the objects were initialized at the same block scope. Now it has no use anymore. I will remove the it.